src/test/metrics.test.ts

```diff
@@ -601,9 +602,124 @@ test('createMetricsData should include extended metrics', (t) => {
   t.truthy(result.platformName);
   t.truthy(result.platformVersion);
   t.true(result.yggdrasilVersion === null);
   t.true(result.specVersion === SUPPORTED_SPEC_VERSION);
 });
+
+test('createMetricsData should include impactMetrics if provided', (t) => {
+  const url = getUrl();
+  // @ts-expect-error
+  const metrics = new Metrics({ url });
+  metrics.start();
+
+  const impactMetrics: CollectedMetric[] = [
+    {
+      name: 'feature_toggle_used',
+      help: 'tracks toggle usage',
+      type: 'counter',
+      samples: [{ labels: { toggle: 'new-ui' }, value: 3 }],
+    },
+  ];
+
+  const result = metrics.createMetricsData(impactMetrics);
+
+  t.truthy(result.impactMetrics);
+  t.deepEqual(result.impactMetrics, impactMetrics);
+});
+
+test('sendMetrics should include impactMetrics in the payload', async (t) => {
+  const url = getUrl();
+
+  const impactMetricSample: CollectedMetric = {
+    name: 'feature_toggle_used',
+    help: 'tracks toggle usage',
+    type: 'counter',
+    samples: [{ labels: { toggle: 'some-feature' }, value: 5 }],
+  };
+
+  const fakeMetricRegistry = {
+    restore: (_metrics: CollectedMetric[]) => {},
+    collect: () => [impactMetricSample],
+  };
+
+  // @ts-expect-error
+  const metrics = new Metrics({
+    url,
+    metricsInterval: 0,
+    metricRegistry: fakeMetricRegistry,
+  });
+
+  const scope = nock(url)
+    .post('/client/metrics', (body) => {
+      t.deepEqual(body.impactMetrics, [impactMetricSample]);
+      return true;
+    })
+    .reply(200);
+
+  await metrics.sendMetrics();
+  t.true(scope.isDone());
+});
+
+test('sendMetrics should restore impactMetrics on failure', async (t) => {
+  const url = getUrl();
+
+  let restored = false;
+
+  const impactMetricSample: CollectedMetric = {
+    name: 'feature_toggle_used',
+    help: 'tracks toggle usage',
+    type: 'counter',
+    samples: [{ labels: { toggle: 'fail-case' }, value: 1 }],
+  };
+
+  const fakeMetricRegistry = {
+    collect: () => [impactMetricSample],
+    restore: (_data: CollectedMetric[]) => {
+      restored = true;
+    },
+  };
+
+  // @ts-expect-error
+  const metrics = new Metrics({
+    url,
+    metricsInterval: 0,
+    metricRegistry: fakeMetricRegistry,
+  });
+
+  nock(url).post('/client/metrics').reply(500);
+
+  await metrics.sendMetrics();
+  t.true(restored);
+});
+
+test('sendMetrics should not include impactMetrics field when empty', async (t) => {
+  const url = getUrl();
+
+  const fakeMetricRegistry = {
+    collect: () => [],
+    restore: (_: CollectedMetric[]) => {},
+  };
+
+  // @ts-expect-error
+  const metrics = new Metrics({
+    url,
+    metricsInterval: 0,
+    metricRegistry: fakeMetricRegistry,
+  });
+
+  // Inject a single toggle evaluation so that we force a metrics send but without impact metrics
+  metrics.count('toggle-x', true);
+
+  const scope = nock(url)
+    .post('/client/metrics', (body) => {
+      t.false('impactMetrics' in body);
```

Comment:
again I'd move the assertion to the end of the test and only record interaction to a variable here